src/app/components/Dashboard/Dashboard.tsx

```diff
@@ -16,21 +17,28 @@ type DashboardProps = {
   }[]
   onDisplayToggle: (id: string) => void
   onTopicSubmit: (topic: Topic) => void
+  onNeedSubmit: (id: string, need: Need) => void
 }
 
-type DisplayMsgType = '' | 'SHOW_TOPIC_FORM'
+type DisplayMsgType = '' | 'SHOW_TOPIC_FORM' | 'SHOW_NEED_FORM'
 
 export default function Dashboard({
   content,
   onDisplayToggle,
   onTopicSubmit,
+  onNeedSubmit,
 }: DashboardProps): JSX.Element {
   const [displayState, setDisplayState] = useState<DisplayMsgType>('')
-
-  function handleSubmit(topic: Topic) {
+  const [topicFocusId, setTopicFocusId] = useState<string | null>(null)
```

Comment:
I don't see why this state is necessary unless your NeedForm needs to display the Topic title or something, so far you could just pass the `id` directly to the handler

Reply:
My bad, forgot that your Form is outside of this `map`